lib/galaxy/schema/groups.py

```diff
@@ -0,0 +1,80 @@
+from typing import List
+
+from pydantic import Field
+from typing_extensions import Literal
+
+from galaxy.schema.fields import (
+    DecodedDatabaseIdField,
+    ModelClassField,
+)
+from galaxy.schema.schema import Model
+
+GROUP_MODEL_CLASS = Literal["Group"]
+
+
+class GroupIndexResponse(Model):
+    """Response schema for a group."""
+
+    model_class: GROUP_MODEL_CLASS = ModelClassField(GROUP_MODEL_CLASS)
+    id: DecodedDatabaseIdField = Field(
```

Comment:
```suggestion
    id: EncodedDatabaseIdField = Field(
```

Sorry, I was not completely clear on how to use the Encoded/Decoded ID handling in models. For **response** models we always use the `Encoded` version of the DatabaseIdField, but for **request** (body, query, or path parameters) we use the `Decoded` version. This ensures that we validate incoming IDs to be properly encoded and provide the decoded value directly for use on the endpoints without manual decoding involved and also ensures that the serialized models in the response are encoded properly when they are returned.

So in summary:
- Input -> Decoded
- Output -> Encoded